xmtp_mls/src/storage/encrypted_store/db_connection.rs

```diff
@@ -19,37 +26,81 @@ pub type DbConnection = DbConnectionPrivate<sqlite_web::connection::WasmSqliteCo
 // callers should be able to accomplish everything with one conn/reference.
 #[doc(hidden)]
 pub struct DbConnectionPrivate<C> {
-    inner: Arc<Mutex<C>>,
+    // Connection with read-only privileges
+    read: Option<Arc<Mutex<C>>>,
+    // Connection with write privileges
+    write: Arc<Mutex<C>>,
+    // Is any connection (possibly this one) currently in a transaction?
+    transaction_lock: Arc<Mutex<()>>,
```

Comment:
nit: it would be helpful at least for me to explicitly state that this lock is shared among all transactions on a particular DB

Reply:
iirc, this transaction lock seems like a global write lock. 
- if the current connection is in a transaction, then we can access the write connection. 
- If there are no transactions, then we can access the write connection. 
- if there is another transaction, we will block until we can access the write connection.

Is that correct?

Reply:
Yes, I would love to just keep the mutex open on the write connection itself and was my original goal, but with transaction closures taking a provider and getting connection references on their own, that route was impossible without severe lifetime leakage. I had to use a double-mutex pattern to avoid putting a lifetime on anything that implemented a dbconnection.